File: VCTServer/src/main/java/com/daelim/icc/vctserver/auth/controller/UserController.java
package com.daelim.icc.vctserver.auth.controller;

import com.daelim.icc.vctserver.auth.dto.UserDTO;
import com.daelim.icc.vctserver.auth.service.UserService;
import com.fasterxml.jackson.core.JsonProcessingException;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.jetbrains.annotations.NotNull;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Slf4j
@RestController
@RequestMapping("user")
@RequiredArgsConstructor
public class UserController {
    private final UserService service;

    @PostMapping("registration")
    public ResponseEntity<String> registration(@RequestBody @NotNull UserDTO userDTO){
        return service.registration(userDTO);
    }

    @PostMapping("signin")
    public ResponseEntity<String> signIn(@RequestParam(value = "id") String userId,
                                         @RequestParam(value = "pwd") String userPwd){
        return service.signIn(userId, userPwd);
    }

    @DeleteMapping("delete")
    public ResponseEntity<String> delete(@RequestHeader(value = "targetId") String userId){
        return service.delete(userId);
    }
}

File: VCTServer/src/main/java/com/daelim/icc/vctserver/ontask/UserController.kt
//package com.daelim.icc.vctserver.auth.controller
//
//import com.daelim.icc.vctserver.auth.dto.UserDTO
//import com.daelim.icc.vctserver.auth.repository.UserRepository
//import com.daelim.icc.vctserver.constdata.SHA256
//import org.springframework.beans.factory.annotation.Autowired
//import org.springframework.web.bind.annotation.*
//
//@RestController
//@RequestMapping("user")
//class UserController {
//
//    @Autowired
//    private lateinit var repository: UserRepository;
//
//    private val sha256 = SHA256();
//
//    @PostMapping("")
//    fun registration(@RequestBody userdto: UserDTO): String {
//        userdto.userPwd = sha256.encrypt(userdto.userPwd);
//        val user = userdto.initUserRegistration();
//        repository.save(user);
//
//        return "success";
//    }
//
//    @GetMapping("")
//    fun login(): String {
//        return "success";
//    }
//}